reconcile/sql_query.py

```diff
@@ -385,112 +377,151 @@ def process_template(query, image_repository, use_pull_secret=False):
     if output not in supported_outputs:
         raise RuntimeError(f"Output {output} not supported")
 
-    make_command = engine_cmd_map[engine]
-
-    command = make_command(
-        output=output, sqlqueries=query["queries"], recipient=query.get("recipient")
+    # concatinate all query files into a single one
+    command = merge_files_command(
+        directory=CONFIG_MAPS_MOUNT_PATH,
+        file_glob="q*",
+        output_file="/tmp/queries",
     )
+    command += engine_cmd_map[engine](sqlqueries_file="/tmp/queries")
+    command += make_output_cmd(output=output, recipient=query.get("recipient", ""))
 
     template_to_render = JOB_TEMPLATE
     render_kwargs = {
         "JOB_NAME": query["name"],
+        "CONFIG_MAPS_MOUNT_PATH": CONFIG_MAPS_MOUNT_PATH,
         "IMAGE_REPOSITORY": image_repository,
         "SECRET_NAME": query["output_resource_name"],
         "ENGINE": engine,
         "ENGINE_VERSION": query["engine_version"],
         "DB_CONN": query["db_conn"],
+        "CONFIG_MAPS": config_map_names,
         "COMMAND": command,
     }
     if use_pull_secret:
         render_kwargs["PULL_SECRET"] = query["name"]
-    if output == "encrypted":
-        render_kwargs["GPG_KEY"] = query["name"]
+
     schedule = query.get("schedule")
     if schedule:
         template_to_render = CRONJOB_TEMPLATE
         render_kwargs["SCHEDULE"] = schedule
 
     template = jinja2.Template(template_to_render)
-    job_yaml = template.render(**render_kwargs)
+    job_yaml = template.render(render_kwargs)
     return job_yaml
 
 
-def openshift_apply(dry_run, oc_map, query, resource):
-    openshift_base.apply(
-        dry_run=dry_run,
-        oc_map=oc_map,
-        cluster=query["cluster"],
-        namespace=query["namespace"]["name"],
-        resource_type=resource.kind,
-        resource=resource,
-        wait_for_namespace=False,
-    )
+def get_config_map(name: str, data: dict) -> dict[str, Any]:
+    return {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {
+            "name": name,
+        },
+        "data": data,
+    }
+
+
+def split_long_query(q, size) -> list[str]:
+    return [q[i : i + size] for i in range(0, len(q), size)]
 
 
-def openshift_delete(dry_run, oc_map, query, resource_type, enable_deletion):
-    try:
-        openshift_base.delete(
+def merge_files_command(directory, file_glob, output_file):
+    return f"cat ''{directory}''/{file_glob} > ''{output_file}'';"
```

Comment:
nit: It is a bit confusing that this command ends with `;` whereas the db commands do not and rely in other functions to do it... I would make the concatenation of commands via the `;` in the `process_template`

Reply:
good point 👍